pkg/liquidity-mining/contracts/admin/GaugeAdder.sol

```diff
@@ -43,143 +48,136 @@ contract GaugeAdder is IGaugeAdder, SingletonAuthentication, ReentrancyGuard {
-     * This function must be called with the address of the *root* gauge which is deployed on Ethereum mainnet.
-     * It should not be called with the address of the gauge which is deployed on Polygon
-     */
-    function addPolygonGauge(address rootGauge) external override authenticate {
-        _addGauge(rootGauge, GaugeType.Polygon);
+    /// @inheritdoc IGaugeAdder
+    function isGaugeFromValidFactory(address gauge, string memory gaugeType)
+        external
+        view
+        override
+        withValidGaugeType(gaugeType)
+        returns (bool)
+    {
+        return _isGaugeFromValidFactory(gauge, gaugeType);
     }
 
-    /**
-     * @notice Adds a new gauge to the GaugeController for the "Arbitrum" type.
-     * This function must be called with the address of the *root* gauge which is deployed on Ethereum mainnet.
-     * It should not be called with the address of the gauge which is deployed on Arbitrum
-     */
-    function addArbitrumGauge(address rootGauge) external override authenticate {
-        _addGauge(rootGauge, GaugeType.Arbitrum);
-    }
+    // Admin Functions
 
-    /**
-     * @notice Adds a new gauge to the GaugeController for the "Optimism" type.
-     * This function must be called with the address of the *root* gauge which is deployed on Ethereum mainnet.
-     * It should not be called with the address of the gauge which is deployed on Optimism.
-     */
-    function addOptimismGauge(address rootGauge) external override authenticate {
-        _addGauge(rootGauge, GaugeType.Optimism);
-    }
+    /// @inheritdoc IGaugeAdder
+    function addGaugeType(string memory gaugeType) external override authenticate {
+        uint256 gaugeTypeLength = bytes(gaugeType).length;
+        require(
+            gaugeTypeLength > 0 && gaugeTypeLength <= 32,
+            "Input string should be between 1 and 32 characters long"
+        );
+        require(!_isValidGaugeType(gaugeType), "Gauge type already added");
 
-    /**
-     * @notice Adds a new gauge to the GaugeController for the "Gnosis" type.
-     * This function must be called with the address of the *root* gauge which is deployed on Ethereum mainnet.
-     * It should not be called with the address of the gauge which is deployed on Gnosis Chain.
-     */
-    function addGnosisGauge(address rootGauge) external override authenticate {
-        _addGauge(rootGauge, GaugeType.Gnosis);
-    }
+        _gaugeTypes.push(gaugeType);
 
-    /**
-     * @notice Adds a new gauge to the GaugeController for the "ZKSync" type.
-     * This function must be called with the address of the *root* gauge which is deployed on Ethereum mainnet.
-     * It should not be called with the address of the gauge which is deployed on ZKSync.
-     */
-    function addZKSyncGauge(address rootGauge) external override authenticate {
-        _addGauge(rootGauge, GaugeType.ZKSync);
+        emit GaugeTypeAdded(gaugeType, gaugeType, _ETHEREUM_TYPE_GAUGE);
```

Comment:
```suggestion
        emit GaugeTypeAdded(gaugeType, gaugeType);
```

We no longer care about this. This value will be emitted by the GaugeController anyway.